gama/configuration/classification_task/classifiers.py

```diff
@@ -0,0 +1,336 @@
+    >>>    )
+
+    voila! 🎉 The `C` hyperparameter is now added to the LogisticRegression classifier
+    with the shared value available in the `shared_hyperparameters` property.
+
+    How to name my hyperparameters ✍️
+    ------------------------------
+
+    The hyperparameters you add to the config space should be named in the following
+    format if similar hyperparameters names can be found in other classifiers:
+
+    >>>    <hyperparameter_name>__<classifier_name>
+
+    For example, the `C` hyperparameter for the `LogisticRegression` classifier should
+    be named `C__logistic_regression` given that the `C` hyperparameter is also
+    available in the `LinearSVC` classifier. This naming convention is used to ensure
+    that the hyperparameters are added to the correct classifier in the config space.
+
+    If your hyperparameter name is unique to your classifier, you can name it as you
+    please without the need to have `__<classifier_name>` at the end of the name.
+    Nonetheless, following the naming convention would in any way not cause any issues.
+
+    Modify an existing classifier 💅
+    -------------------
+
+    To modify an existing classifier, adjust its respective setup method and the
+    shared hyperparameters property as needed by modifying the values of the
+    hyperparameters. For example, to change the value of the `C` hyperparameter for
+    the `LogisticRegression` classifier, change the value of the `C` hyperparameter
+    in the `shared_hyperparameters` property by:
+
+    >>>    "C": [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1.0, 5.0, 10.0, 15.0, 20.0, 25.0],
+
+    The `C` hyperparameter will then be added to the config space with the appropriate
+    value. However, be cautious, if you change values in the shared hyperparameters
+    property, it will be changed for all classifiers that use that hyperparameter.
+    If you want this change to only apply to a specific classifier, you should add
+    the hyperparameter to the setup method of that classifier. E.g. if you want to
+    change the value of the `C` hyperparameter for the `LogisticRegression` classifier,
+    and only want this change to apply to the `LogisticRegression` classifier, add
+    the following line to the `setup_logistic_regression` method:
+
+    >>>    C = csh.CategoricalHyperparameter(
+    >>>        "C__logistic_regression",
+    >>>        [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1.0, 5.0, 10.0, 15.0, 20.0, 25.0]
+    >>>    )
+
+    The `C` hyperparameter will be added as-is for the `LogisticRegression` classifier
+    and the value of the `C` hyperparameter for other classifiers will be as available
+    in the `shared_hyperparameters` property – iff they use the `C`
+    hyperparameter of the `shared_hyperparameters` property.
+
+    Parameters
+    ----------
+    config_space : cs.ConfigurationSpace
+        The ConfigSpace object that defines the hyperparameters and their ranges for
+        the classifiers.
+
+    """
```

Comment:
I think this should also go in the configuration documentation. It's going to be easier for people to find, and it can centralize all the information.

Reply:
Great ! Will be done in January, following the full review.